libs/mf-tools/src/lib/web-components/web-component-wrapper.ts

```diff
@@ -40,29 +48,35 @@ export class WebComponentWrapper implements AfterContentInit, OnChanges {
   }
 
   private populateProps() {
+    this.props = this.props ?? this.options.props;
     for (const prop in this.props) {
       this.element[prop] = this.props[prop];
     }
   }
 
   private setupEvents() {
+    this.events = this.events ?? this.options.events;
     for (const event in this.events) {
       this.element.addEventListener(event, this.events[event]);
     }
   }
 
   async ngAfterContentInit() {
-    const options =
+    this.options =
       this.options ?? (this.route.snapshot.data as WebComponentWrapperOptions);
 
     try {
-      await loadRemoteModule(options);
-
-      this.element = document.createElement(options.elementName);
+      await loadRemoteModule(this.options);
+      this.element = document.createElement(this.options.elementName);
       this.populateProps();
       this.setupEvents();
 
-      this.vc.nativeElement.appendChild(this.element);
+      if (this.options.shadowDom) {
+        const shadow = this.vc.nativeElement.attachShadow({ mode: 'open' });
```

Comment:
Shouldn't you be using `this.options.shadowDom` as parameter to `attachDom` ?

Reply:
Thanks, you are absolutly right. Will fix it tomorrow!